### PR TITLE
Fix NaN in jax.nn.standardize due to negative variance #30426

### LINKED ISSUE #30426: jax.nn.standardize returns nan when variance is lower than epsilon
### Description

Sirs,

   `jax.nn.standardize` could be easily improved by taking a lower bound on variance instead of adding epsilon. I found a minimal reproducible example and hacked a quick fix (below). Your comment in the code admits lower accuracy, but at least it should not return nans unnecessarily, don't you think?

`return jnp.subtract(x, jnp.asarray(mean)) * lax.rsqrt(jnp.asarray(variance) + epsilon)`
replace with
`return jnp.subtract(x, jnp.asarray(mean)) * lax.rsqrt(jnp.maximum(jnp.asarray(variance), epsilon))`

Example:
```
x = -11. * jnp.ones((3,))
noise = jrnd.normal(jrnd.key(0)) * 2e-6
jax.nn.standardize(x + noise)
# returns Array([nan, nan, nan], dtype=float32)
```

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.5.2
jaxlib: 0.5.1
numpy:  2.0.2
python: 3.11.13 (main, Jun  4 2025, 08:57:29) [GCC 11.4.0]
device info: cpu-1, 1 local devices"
process_count: 1
platform: uname_result(system='Linux', node='5453835f9caa', release='6.1.123+', version='#1 SMP PREEMPT_DYNAMIC Sun Mar 30 16:01:29 UTC 2025', machine='x86_64')
```

### PR DESCRIPTION
**Pull request by nandan2003**
## Description
Fixes #30426.

`jax.nn.standardize` previously returned `NaN` when the input array had a variance slightly less than 0 (due to floating point imprecision) but larger in magnitude than `-epsilon`. In these cases, `variance + epsilon` remained negative, causing `rsqrt` to produce NaNs.

This PR changes the normalization logic to clamp the variance using `jnp.maximum(variance, epsilon)` instead of adding epsilon. This ensures the denominator is always valid.

## Reproduction
```python
import jax
import jax.numpy as jnp

# Input that causes slight negative variance due to precision
val = -11.0
x = val * jnp.ones((3,))
noise = jax.random.normal(jax.random.key(0)) * 2e-6
bad_input = x + noise

# Before: Returns [nan, nan, nan]
# After: Returns valid floats
res = jax.nn.standardize(bad_input)
print(res)

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks for this – I don't think this is the correct fix for the issue. See my comment at https://github.com/jax-ml/jax/issues/30426#issuecomment-3572033171

**Comment by nandan2003** [comment]
Closing this PR as the issue has been resolved in main
